### ISSUE TITLE
Fix compilation of languageinfo.cpp with MinGW-TDM 9.2

### ISSUE THREAD

**Issue by vadz** [wxWidgets/wxWidgets#22100]
As discussed in [this thread](https://groups.google.com/g/wx-dev/c/daQonFPaGeM/m/ClM-JGs7AgAJ), at least TDM MinGW 9.2 can't compile `wxUILocale::InitLanguagesDB()` currently. This function has "only" 879 `LNG()` macro invocations, but considering that each of them expands in 9 lines, I suspect that it might be too long for the compiler to deal with it.

One possibility is to split the function into several chunks and this was confirmed to already work. Another, slightly preferable one IMHO if it can works too, would be to just add `AddLanguage()` overload taking all the fields and call it directly without using a macro -- this should divide the number of lines in the function by 9.

@utelle I'd be very grateful if you could please look at this.

@XPsoud If you can check whether implementing the second workaround works too, it would be very nice as well.

TIA!

**Comment by XPsoud** [comment]
Checked, and it works.
Here is the test :

- I've created a "MyAddLanguage" function in src/common/languageinfo.cpp witch takes the 9 parameters for the LNG macro
```cpp
void MyAddLanguage(int wxlang, wxString bcp47tag, wxString canonical, wxString canonicalref, wxUint32 winlang, wxUint32 winsublang, wxLayoutDirection layout, wxString desc, wxString descnative)
{
    wxLanguageInfo info;
    info.Language = wxlang;
    info.LocaleTag = bcp47tag;
    info.CanonicalName = canonical;
    info.CanonicalRef = canonicalref;
    info.LayoutDirection = layout;
    info.Description = desc;
    info.DescriptionNative = descnative;
#ifdef __WIN32__
    info.WinLang = winlang;
    info.WinSublang = winsublang;
#endif // __WIN32__
    wxUILocale::AddLanguage(info);
}
```

- Then,  I've replaced all the `LNG(......)` macros in `wxUILocale::InitLanguagesDB()` by `MyAddLanguage(......);`
- I didn't touched the `SCMAP(...)` macros in this file

As it can be seen, I've only used `wxString` arguments to (quickly) make the tests so there should be some checks regarding the LNG macro, but 'as is', it compiles.

**Comment by vadz** [comment]
Great, thanks a lot, Xavier!

I should be able to modify the script in `misc/languages/genlang.py` to generate code like this instead of the current one myself, but if anybody beats me to, I wouldn't complain :-)

**Comment by utelle** [comment]
Personally, I would try to solve the issue in a different way, namely by generating a static data array containing the language information and then executing a simple loop over the data array filling the language database instead of generating code for each language entry.

**Comment by utelle** [comment]
Most likely I would be able to adjust the Python script accordingly within the next week.

**Comment by vadz** [comment]
I wouldn't mind doing it like this at all, but IMO it doesn't really matter much, I'd just like (all) compilers to be able to compile this.

This is not urgent, so if you can do it in your preferred way next week it would be great, but I'd like to fix this before 3.1.6. TIA!

**Comment by utelle** [comment]
> I wouldn't mind doing it like this at all, but IMO it doesn't really matter much, I'd just like (all) compilers to be able to compile this.

Well, maybe an optimizing compiler can generate code that is almost as compact and efficient, but I wouldn't bet on it.

From the issue description I got the impression that this might be a preprocessor issue. Generating a data array and processing it with a simple loop would completely eliminate the need for the LNG macro.

> This is not urgent, so if you can do it in your preferred way next week it would be great, but I'd like to fix this before 3.1.6. TIA!

I fully agree that the issue should be solved before 3.1.6. As said I'll try to address it within the upcoming week.

BTW, if possible PR #2615 should also be addressed before 3.1.6.

**Comment by utelle** [comment]
PR #22104 should fix the issue.
